datadog_checks_base/datadog_checks/base/utils/db/statement_metrics.py

```diff
@@ -0,0 +1,127 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class StatementMetrics:
+    """
+    This class supports normalized statement-level metrics, which are collected from the database's
+    statistics tables, ex:
+
+        - Postgres: pg_stat_statements
+        - MySQL: performance_schema.events_statements_summary_by_digest
+        - Oracle: V$SQLAREA
+        - SQL Server: sys.dm_exec_query_stats
+        - DB2: mon_db_summary
+
+    These tables are monotonically increasing.
+    """
+
+    def __init__(self):
+        self.previous_statements = {}
```

Comment:
(Nit) Is this meant to be accessed from the outside? If not, let's consider turning it into a private attribute:

```suggestion
        self._previous_statements = {}
```